### PR TITLE
httplib with OpenSSL mock support

### PR DESCRIPTION
**Pull request by andresriancho**
Done:
- Added httplib with OpenSSL mock support
- Tested with test_openssl.py (all PASS)

Running all functional tests (at least on my workstation) fails with some "port already in use" errors. What's that about? Any ideas on how to fix that? (it was a problem even before this patch)

I'm sending this PR with two main objectives:
 1- Get travis to run all tests (maybe there we don't get those errors)
 2- Get an initial review of the code/changes

### CODE REVIEW

**Comment by andresriancho** [comment]
Aha! Now these builds tell me something I didn't know :)
 * https://travis-ci.org/gabrielfalcao/HTTPretty/jobs/49458114
 * https://travis-ci.org/gabrielfalcao/HTTPretty/jobs/49458113

Will fix in a while

**Comment by andresriancho** [comment]
@gabrielfalcao there are two things to notice about the builds:

 * Some errors in the functional test which seem to be unrelated with my patch https://travis-ci.org/gabrielfalcao/HTTPretty/builds/49459191
 * Functional test run looks like `HTTPretty should all a callback function to be set as the body with httplib2 ... (10289, 200000)` , what's `(10289, 200000)`?

**Comment by lithiumFlower** [comment]
Any chance this will get merged?

**Comment by gabrielfalcao** [comment]
Hi!
I'm going through all the open PR's in HTTPretty and merging anything that fulfills these conditions:

* Travis CI passing
* Have test coverage

Your PR has test coverage but the travis build is breaking, I'd love to merge this as soon as the travis build is passing.

Thoughts?

**Comment by va3093** [comment]
@andresriancho This implementation won't work for python 3 since `socket._fileobject` is removed. This PR may be helpful in fixing that https://github.com/pallets/werkzeug/pull/565/files

File: httpretty/core.py [hunk 1]
```diff
@@ -468,7 +490,28 @@ def fake_wrap_socket(s, *args, **kw):
     return s
 
 
-def create_fake_connection(address, timeout=socket._GLOBAL_DEFAULT_TIMEOUT, source_address=None):
+def fake_fileobject(s, *args, **kwargs):
```
Comment (gabrielfalcao) [inline httpretty/core.py]: this is nice

**Pull request closed without merging**